sql/sql_trigger.cc

```diff
@@ -2006,7 +2006,7 @@ bool Table_triggers_list::change_table_name(THD *thd, const char *db,
   LEX_STRING *err_trigname;
   DBUG_ENTER("change_table_name");
 
-  bzero(&table, sizeof(table));
+  bzero(static_cast<void*>(&table), sizeof(table));
```

Comment:
This does not fix the root cause of the warning. If the actual fix is too complex for us at this point of 5.5 lifecycle, just patch the build for -Wno-class-memaccess instead.